### ISSUE TITLE
unable to use testcontainers with credentials (ryuk + private registry)

### ISSUE THREAD

**Issue by peternitro** [testcontainers/testcontainers-java#813]
Recently upgraded testcontainers from 1.4.3 to 1.8.1 and found that it is unable to pull images from our private registry at docker hub. The daemon is configured with an auth token created by `docker login` and persisted in `~/.docker/config.json`. Testcontainers/dockerjava appears to ignore the `auth` tokens persisted there?

To unblock our Jenkins plugins I've added a `~/.docker-java.properties` with the registry & credentials for our private docker hub regristry as per https://github.com/docker-java/docker-java. This allowed images to be pulled down again. 

However, when running the first time, testcontainers pulls down ryuk which fails as follows:

```
2018-08-01 14:45:03,581 [test] INFO  o.t.DockerClientFactory - Docker host IP address is localhost
2018-08-01 14:45:03,836 [test] INFO  o.t.DockerClientFactory - Connected to docker: 
  Server Version: 1.10.3
  API Version: 1.22
  Operating System: Ubuntu 14.04.4 LTS
  Total Memory: 16047 MB
2018-08-01 14:45:04,656 [test] ERROR c.g.d.c.a.ResultCallbackTemplate - Error during callback
com.github.dockerjava.api.exception.InternalServerErrorException: Get https://quay.io/v2/testcontainers/ryuk/manifests/0.2.2: unauthorized: Invalid Username or Password

	at com.github.dockerjava.netty.handler.HttpResponseHandler.channelRead0(HttpResponseHandler.java:109)
	at com.github.dockerjava.netty.handler.HttpResponseHandler.channelRead0(HttpResponseHandler.java:33)
	at org.testcontainers.shaded.io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at org.testcontainers.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at org.testcontainers.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
```
It looks like it's trying to use the configured credentials to access the public repo at quay.io and of course fails.  Chicken and egg situation?

**Comment by peternitro** [comment]
related: https://github.com/testcontainers/testcontainers-java/issues/790

**Comment by peternitro** [comment]
fixed by virtue of fixing #820
